cabal-install/Distribution/Client/Dependency.hs

```diff
@@ -346,7 +348,7 @@ dontUpgradeNonUpgradeablePackages params =
   where
     extraConstraints =
       [ LabeledPackageConstraint
-        (PackageConstraintInstalled pkgname)
+        (PackageConstraint (unqualified pkgname) PackagePropertyInstalled)
```

Comment:
Yes I see your point. This may require a little thought, so I'll make a new PR after this one. Would it make sense instead of expanding `PackageConstraint` to expand `Qualifier` by adding a new constructor? I see two obvious choices:

1. Let 'Unqualified' mean top-level dependency and add a new constructor called e.g. `Any`.
2. Let 'Unqualified' mean applies everywhere and add a new constructor `Toplevel`.

Option 1 seems somewhat confusing semantically, so maybe option 2 is better - what do you think?